documentation/en/chain-indexer-overview-for-operators.md

```diff
@@ -0,0 +1,415 @@
+
+## Downgrade Steps
+
+In case you need to downgrade to the [previous indexing system](#previous-indexing-system), follow these steps:
+
+1. Prevent the node from receiving traffic.
+2. Stop your Lotus node.
+3. Download or build a Lotus binary for the rollback version which has the implementation of the old `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices.
+4. Ensure that you've set the correct config for the existing `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices in the `config.toml` file.
+5. Restart your Lotus node.
+6. Backfill the `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices using the `lotus-shed index backfill-*` CLI tooling available in the [previous indexing system](#previous-indexing-system) for the range of epochs between the upgrade to `ChainIndexer` and the rollback of `ChainIndexer`.
+7. Route traffic back to the node.  
+
+## Terminology
+### Previous Indexing System
+* This corresponds to the indexing system used in Lotus versions before v1.31.0.  
+* It has been replaced by the [ChainIndexer](#chainindexer-indexing-system).
+* It was composed of three indexers using three separate databases: [`EthTxHashLookup`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/ethhashlookup/eth_transaction_hash_lookup.go), [`MsgIndex`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/index/msgindex.go), and [`EventIndex`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/events/filter/index.go).
+* It persisted state to the [removed option](#removed-options) for `Events.DatabasePath`, which defaulted to `${LOTUS_PATH}/sqlite`.
+* It had CLI backfill tooling: `lotus-shed index backfill-*`
+
+### ChainIndexer Indexing System
+* This corresponds to the indexing system used in Lotus versions v1.31.0 onwards.  
+* It replaced the [previous indexing system](#previous-indexing-system).
+* It is composed of a single indexer, [`ChainIndexer`](https://github.com/filecoin-project/lotus/blob/master/chain/index/indexer.go), using a [single database for transactions, messages, and events](https://github.com/filecoin-project/lotus/blob/master/chain/index/ddls.go).
+* It persists state to `${LOTUS_PATH}/chainindex`.
+* It has this CLI backfill tooling: [`lotus-shed chainindex validate-backfill`](#lotus-shed-chainindex-validate-backfill-cli-tool)
+* **Storage requirements:** See the [backfill disk space requirements](#backfill-disk-space-requirements).
+* **Backfil times:** See the [backfill timing](#backfill-timing).
+
+## Appendix
+
+### Wny isn't there an automated migration from the [previous indexing system](#previous-indexing-system) to the [ChainIndexer indexing system](#chainindexer-indexing-system)?
```

Comment:
Why*

Reply:
Fixed in https://github.com/filecoin-project/lotus/pull/12600/commits/8463a03345d8c28410825d0cc115224976522d31 .  Resolving.